GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -857,7 +857,7 @@ public static string CleanUpCmd(bool dryrun, bool directories, bool nonignored,
         public static string GetAllChangedFilesCmd(bool excludeIgnoredFiles, UntrackedFilesMode untrackedFiles, IgnoreSubmodulesMode ignoreSubmodules = IgnoreSubmodulesMode.None, bool noLocks = false)
         {
             StringBuilder stringBuilder = new StringBuilder( 
-                noLocks && VersionInUse.SupportNoOptionalLocks ? "--no-optional-locks" : ""
+                (noLocks && VersionInUse.SupportNoOptionalLocks ? "--no-optional-locks " : "")
```

Comment:
Normally, we do not want extra spaces like this, but this is done better in master, using the ArgumentBuilder.